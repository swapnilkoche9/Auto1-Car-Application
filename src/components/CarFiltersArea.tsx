import React from 'react'
import Dropdown from './Dropdown'
import Button from './Button'
import constants from '../constants/SystemConstants'
import {CarFiltersAreaProps} from '../utils/interface'


const CarFiltersArea = (props:CarFiltersAreaProps) => {
  const {
    colors,
    manufacturers,
    getColorsFilterParams,
    getManufacturersFilterParams,
    getFilteredCarList
  } = props

  return (
    <div className='filtersContainer'>
      <div className='filtersContainerContent'>
        <div className='filterDropdown'>
          <Dropdown
            dropdownContent={colors}
            defaultDropdownValue={constants.ALL_CAR_COLORS_DROPDOWN}
            getFilterParams={getColorsFilterParams}
            dropDownLabel={constants.COLOR}
          />
        </div>
        <div className='filterDropdown'>
          <Dropdown
            dropdownContent={manufacturers}
            defaultDropdownValue={constants.ALL_MANUFACTURERS_DROPDOWN}
            getFilterParams={getManufacturersFilterParams}
            dropDownLabel={constants.MANUFACTURERS}
          />
        </div>
        <div className='filterBtnContainer'>
          <Button handleClick={getFilteredCarList} text={constants.BUTTON_TEXT} />
        </div>
      </div>
    </div>
  )
}

export default CarFiltersArea
