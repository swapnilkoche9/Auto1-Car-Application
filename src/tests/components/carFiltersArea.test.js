import React from 'react'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import CarFiltersArea from '../../components/CarFiltersArea'

Enzyme.configure({ adapter: new Adapter() })
const { shallow } = Enzyme

const getColorsFilterParams = jest.fn()
const getManufacturersFilterParams = jest.fn()
const getFilteredCarList = jest.fn()
const wrapper = shallow(
  <CarFiltersArea
    colors={['Yellow']}
    manufacturers={['BMW']}
    getColorsFilterParams={getColorsFilterParams}
    getManufacturersFilterParams={getManufacturersFilterParams}
    getFilteredCarList={getFilteredCarList}
  />
)
test('should render CarFiltersArea correctly', () => {
  expect(wrapper).toMatchSnapshot()
})

test('should render CarFiltersArea with alt data correctly', () => {
  wrapper.setProps({
    totalPageCount: 5
  })
  expect(wrapper).toMatchSnapshot()
})
