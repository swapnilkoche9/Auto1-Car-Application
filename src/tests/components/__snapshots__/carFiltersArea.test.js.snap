// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CarFiltersArea correctly 1`] = `ShallowWrapper {}`;

exports[`should render CarFiltersArea with alt data correctly 1`] = `ShallowWrapper {}`;
